nightmare/paths/graphs.spec.js

```diff
@@ -0,0 +1,146 @@
+        .then(() => { done(); })
+        .catch(catchErrors(done));
+      })
+      .then(() => { done(); })
+      .catch(catchErrors(done));
+    });
+
+    it('should display Bar chart graph in page', (done) => {
+      nightmare
+      .evaluate(() => document.querySelector('.page-viewer.document-viewer > div.alert.alert-info a').href)
+      .then(link => nightmare.goto(link))
+      .then((page) => {
+        expect(page.code).toBe(200);
+        return nightmare
+        .wait('#app > div.content > div > div > main div.markdown-viewer')
+        .getInnerHtml('#app > div.content > div > div > main div.markdown-viewer')
+        .then((html) => {
+          expect(html).toContain('class="BarChart "');
+          expect(html).toContain('class="recharts-responsive-container"');
+        });
+      })
+      .then(() => { done(); })
+      .catch(catchErrors(done));
+    });
+
+    it('should navigate back to edit page', (done) => {
+      nightmare
+      .waitToClick(selectors.navigation.settingsNavButton)
+      .wait(selectors.settingsView.settingsHeader)
+      .waitToClick(localSelectors.pagesButton)
+      .wait('#app > div.content > div > div > div.settings-content > div.panel-default > ul.pages')
+      .wait(500)
+      .evaluate(() => document.querySelector('div.settings-content > div.panel-default > ul.pages > li:nth-child(1) > a').text)
+      .then((pageTitle) => {
+        expect(pageTitle).toBe('Page data viz');
+        return nightmare.evaluate(() => document.querySelector('div.settings-content > div.panel-default > ul.pages > li:nth-child(1) > a').href);
+      })
+      .then((link) => {
+        expect(link).toContain('/pages/edit/');
+        nightmare.goto(link);
+        done();
+      })
+      .catch(catchErrors(done));
+    });
+
+    it('should insert Pie chart graph in page', (done) => {
+      nightmare
+      .evaluate(() => document.querySelector('div.panel-body.page-viewer.document-viewer > div > div.tab-content.tab-content-visible > textarea').value)
+      .then((input) => {
+        expect(input).toContain('<Dataset />');
+
+        return nightmare
+        .write(localSelectors.pageContentsInput, graphs.pieChart)
+        .waitToClick(localSelectors.savePageButton)
+        .wait('div.panel-body.page-viewer.document-viewer > div.alert.alert-info:first-of-type')
+        .getInnerText('div.panel-body.page-viewer.document-viewer > div.alert.alert-info:first-of-type')
+        .then((text) => {
+          expect(text).toContain('/page');
+          expect(text).toContain('(view page)');
+        })
+        .then(() => { done(); })
+        .catch(catchErrors(done));
+      })
+      .then(() => { done(); })
+      .catch(catchErrors(done));
+    });
+
+    it('should display Pie chart graph in edited page', (done) => {
+      nightmare
+      .evaluate(() => document.querySelector('div.panel-body.page-viewer.document-viewer > div.alert.alert-info a').href)
+      .then(link => nightmare.goto(link))
```

Comment:
@elric-wamugu why an evaluate to get href and goto instead of clicking the anchor itself ?